service/service_test.go

```diff
@@ -511,6 +518,105 @@ func TestApplication_GetExporters(t *testing.T) {
 	<-appDone
 }
 
+func TestSetFlag(t *testing.T) {
+	factories, err := defaultcomponents.Components()
+	require.NoError(t, err)
+	params := Parameters{
+		Factories: factories,
+	}
+	t.Run("unknown_component", func(t *testing.T) {
```

Comment:
@tigrannajaryan have a look at this test.

If the flag references an unknown property the factory returns an error. However, if the flag references a valid component that is not in the config file e.g. `batch/foo` then an error is not returned (see the next test) and a new component instance is created that is not enabled in the pipeline. I think that this behaviour is fine.

Reply:
>However, if the flag references a valid component that is not in the config file e.g. batch/foo then an error is not returned (see the next test) and a new component instance is created that is not enabled in the pipeline. I

This is the part that I don't like. If I have a typo in my `--set` flag and the name of the component is incorrect then it will have no effect and there will be no indication to the user that something went wrong.

Preferably in this case the startup should fail and tell that the component is not found. Is it possible to do?

Reply:
I cannot think of a simple solution. The same can happen with the config file a user can forget to add a component to a pipeline.

The ugly solution might be to parse the config in `FileLoaderConfigFactory` then pass it into `AddSetFlagProperties` which would parse it's own config and compare which components are not in the pipeline. Then parse the final config at the end of `FileLoaderConfigFactory` with the final viper. This is a rather horrible solution :).

Maybe we could print a warning for not assigned components during the config validation?